superset/tags/models.py

```diff
@@ -179,8 +179,8 @@ def after_insert(
             tag_id=tag.id, object_id=target.id, object_type=cls.object_type
         )
         session.add(tagged_object)
-
         session.commit()
+        session.close()
```

Comment:
The various SQLAlchemy hooks, `after_insert`, `after_delete`, etc. pass in the connection from which we instantiate a new session which was never closed. I'm not entirely sure why we're not using the Flask-SQLAlchemy session here (and not committing) to ensure the DAO et al. logic is captured in a single transaction.